operator/src/main/java/oracle/kubernetes/operator/helpers/RollingHelper.java

```diff
@@ -208,31 +208,18 @@ public NextAction apply(Packet packet) {
 
       LOGGER.info(MessageKeys.ROLLING_SERVERS, dom.getDomainUid(), servers, readyServers);
 
-      int countToRestartNow = Math.max(1, countReady - dom.getMinAvailable(clusterName));
+      int countToRestartNow = countReady - dom.getMinAvailable(clusterName);
       Collection<StepAndPacket> restarts = new ArrayList<>();
       for (int i = 0; i < countToRestartNow; i++) {
-        restarts.add(new StepAndPacket(new RestartOneClusteredServerStep(servers, null), packet));
+        StepAndPacket serverToRestart = servers.poll();
```

Comment:
Functional programming structures can be much easier to read. In this case, 

```
   Optional.ofNullable(servers.poll()).ifPresent(serverToRestart -> restarts.add(serverToRestart));
```

makes it clearer that you are adding the result of the call to servers.poll() to the collection when it exists. It takes some getting used to, but the result is easier-to-read code.

Reply:
Updated with your functional programming structure.